README.md

```diff
@@ -273,27 +273,27 @@ assert_eq!(None, it.next());
 ```
 
 And, again, no magic in sight. This is effectively just syntactic sugar.  
-In fact, it's all so _not_ magic that the compiler did not even realize that our combinator chain is absolute non-sense and a complete waste of resources:
+In fact, it's all so _not_ magic that the compiler did not even realize that our combinator chain is absolute nonsense and a complete waste of resources:
 ```sh
 $ cargo rustc --lib -- --test -Zprint-type-sizes
 # [...]
 `Bounds<Bounds<Bounds<Range<usize>, usize>, usize>, usize>`: 72 bytes, alignment: 8 bytes
 ```
-How could it? It's just blindly monomorphizing structs inside of other structs, that's all there is to it!
+How could it? It's just blindly monomorphizing structs inside of other structs; that's all there is to it!
 
 **What we have so far**
 
-We can now build up complex state-machines by composing them from smaller, simpler parts... and what's more, we can even do it in an expressive, readable and maintainable way.
+We can now build up complex state machines by composing them from smaller, simpler parts... and what's more, we can even do it in an expressive, readable and maintainable way.
```

Comment:
I didn't make the change, but I personally prefer to use the "Oxford comma":
... in an expressive, readable, and maintainable way.

Dealer's choice, though.